app/src/main/java/org/thoughtcrime/securesms/video/videoconverter/VideoTrackConverter.java

```diff
@@ -0,0 +1,512 @@
+        final int outputHeightRotated;
+        if ((rotation % 180 == 90)) {
+            //noinspection SuspiciousNameCombination
+            outputWidthRotated = outputHeight;
+            //noinspection SuspiciousNameCombination
+            outputHeightRotated = outputWidth;
+        } else {
+            outputWidthRotated = outputWidth;
+            outputHeightRotated = outputHeight;
+        }
+
+        final MediaFormat outputVideoFormat = MediaFormat.createVideoFormat(videoCodec, outputWidthRotated, outputHeightRotated);
+
+        // Set some properties. Failing to specify some of these can cause the MediaCodec
+        // configure() call to throw an unhelpful exception.
+        outputVideoFormat.setInteger(MediaFormat.KEY_COLOR_FORMAT, MediaCodecInfo.CodecCapabilities.COLOR_FormatSurface);
+        outputVideoFormat.setInteger(MediaFormat.KEY_BIT_RATE, videoBitrate);
+        outputVideoFormat.setInteger(MediaFormat.KEY_FRAME_RATE, OUTPUT_VIDEO_FRAME_RATE);
+        outputVideoFormat.setInteger(MediaFormat.KEY_I_FRAME_INTERVAL, OUTPUT_VIDEO_IFRAME_INTERVAL);
+        if (VERBOSE) Log.d(TAG, "video format: " + outputVideoFormat);
+
+        // Create a MediaCodec for the desired codec, then configure it as an encoder with
+        // our desired properties. Request a Surface to use for input.
+        final AtomicReference<Surface> inputSurfaceReference = new AtomicReference<>();
+        mVideoEncoder = createVideoEncoder(videoCodecInfo, outputVideoFormat, inputSurfaceReference);
+        mInputSurface = new InputSurface(inputSurfaceReference.get());
+        mInputSurface.makeCurrent();
+        // Create a MediaCodec for the decoder, based on the extractor's format.
+        mOutputSurface = new OutputSurface();
+
+        mOutputSurface.changeFragmentShader(createFragmentShader(
+                inputVideoFormat.getInteger(MediaFormat.KEY_WIDTH), inputVideoFormat.getInteger(MediaFormat.KEY_HEIGHT),
+                outputWidth, outputHeight));
+
+        mVideoDecoder = createVideoDecoder(inputVideoFormat, mOutputSurface.getSurface());
+
+        mVideoDecoderInputBuffers = mVideoDecoder.getInputBuffers();
+        mVideoEncoderOutputBuffers = mVideoEncoder.getOutputBuffers();
+        mVideoDecoderOutputBufferInfo = new MediaCodec.BufferInfo();
+        mVideoEncoderOutputBufferInfo = new MediaCodec.BufferInfo();
+
+        if (mTimeFrom > 0) {
+            mVideoExtractor.seekTo(mTimeFrom * 1000, MediaExtractor.SEEK_TO_PREVIOUS_SYNC);
+            Log.i(TAG, "Seek video:" + mTimeFrom + " " + mVideoExtractor.getSampleTime());
+        }
+    }
+
+    void setMuxer(final @NonNull Muxer muxer) throws IOException {
+        mMuxer = muxer;
+        if (mEncoderOutputVideoFormat != null) {
+            Log.d(TAG, "muxer: adding video track.");
+            mOutputVideoTrack = muxer.addTrack(mEncoderOutputVideoFormat);
+        }
+    }
+
+    void step() throws IOException, TranscodingException {

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 52. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 195-347 into a separate method.